### DISCUSSION TITLE
Is there a way to reduce the Lottie file size?

### DISCUSSION THREAD

**Discussion by lucksp** [lottie-react-native/lottie-react-native#884] [Q&A]
I am doing a little research on whether or not we can compress Lottie animation files to help reduce bundle size for our mobile app. 
It appears Lottie files are JSON format.  Is this something that can be compressed?  Or does the animation source really need to control how detailed the animations are and therefore the JSON output is reduced?

Any other suggestions for how to reduce assets size?

Does this library work well with [dotLottie](https://www.dotlottie.io/)?

**Comment by TheRogue76** [comment]
Hi @lucksp,

Yes, It does! We added support in Version 6.1.0. You can read the new section on how to use DotLottie [here](https://github.com/lottie-react-native/lottie-react-native?tab=readme-ov-file#if-you-want-to-use-lottie-files). DotLottie is actually our preferred format now, since unlike the JSON variant that is auto-loaded on the JS side and then sent over the bridge (A limitation that we can not fix due to how it is the default behavior in metro), we load Dotlottie on the native sides, so it is both lighter and much quicker and with less headache.

**Reply by taenri** [reply]
Hi @TheRogue76 - My team is integrating lottie-react-native for usage in our C++ Windows application. Currently, the README states that Dynamic loading of Lottie JSON strings is only supported in C# Windows applications. Is it possible to use DotLottie archives for CodeGen scenarios at all? We currently are running into an issue during the CodeGen steps with some of our larger animations where the build pipeline VM keeps running out of memory during the linking process:
`C1002: compiler is out of heap space in pass 2`
